File: backend/patient-service/src/main/java/org/example/patientservice/service/PatientService.java
package org.example.patientservice.service;

import org.example.patientservice.dto.CreatePatientRequest;
import org.example.patientservice.dto.PatientDto;
import org.example.patientservice.entity.Patient;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Optional;

public interface PatientService {
    PatientDto createPatient(PatientDto patientDto);

    List<PatientDto> getAllPatients();

    PatientDto getPatientById(Integer patientId);

    PatientDto updatePatient(Integer patientId, PatientDto patientDto);

    void deletePatient(Integer patientId);

    Optional<PatientDto> searchPatient(String identityNumber, String insuranceNumber, String fullName, String email, String phone);

    PatientDto getPatientByUserId(Integer userId);

    PatientDto registerPatient(CreatePatientRequest request);

    PatientDto uploadAvatar(Integer id, MultipartFile file);

    PatientDto deleteAvatar(Integer id);
}

File: frontend/web/src/services/patientService.ts
import { api } from "./api";
import { Patient, PatientDto } from "../types/patient";

export const patientService = {
  async getAllPatients(): Promise<Patient[]> {
    const response = await api.get<Patient[]>("/patients");
    return response.data;
  },

  async getPatientById(patientId: string): Promise<Patient> {
    const response = await api.get<Patient>(`/patients/${patientId}`);
    return response.data;
  },

  async createPatient(patientData: PatientDto): Promise<Patient> {
    const response = await api.post<Patient>("/patients", patientData);
    return response.data;
  },

  async updatePatient(
    patientId: string,
    patientData: Partial<PatientDto>
  ): Promise<Patient> {
    const response = await api.put<Patient>(
      `/patients/${patientId}`,
      patientData
    );
    return response.data;
  },

  async deletePatient(patientId: string): Promise<string> {
    const response = await api.delete<string>(`/patients/${patientId}`);
    return response.data;
  },

  async searchPatient(params: {
    identityNumber?: string;
    insuranceNumber?: string;
    fullName?: string;
  }): Promise<Patient | null> {
    const query = new URLSearchParams(params).toString();
    const response = await api.get<Patient | null>(`/patients/search?${query}`);
    return response.data;
  },
};
